### DISCUSSION TITLE
Recette V5.4

### DISCUSSION THREAD

**Discussion by alexisig** [MTES-MCT/sparte#332] [Recettes]
Recette V5.4 - Nouveau millésime CEREMA.

Changelog : https://github.com/MTES-MCT/sparte/blob/3e29aa6fc134d94abcf880d1badecfa2af1767d7/CHANGELOG.md
Environnement pour tester : https://sparte-staging-pr321.osc-fr1.scalingo.io/

**IMPORTANT** : les tests sont uniquement possibles pour le Gers ou pour les départements n'ayant jamais reçu d'OCS GE. Nous ne chargeons pas toutes les données OCS GE sur les environnements de test afin de pouvoir les rendre disponibles plus rapidement.

---

Pour les bugs déjà présents en production : faire une issue sur le dépôt github https://github.com/MTES-MCT/sparte/issues
Pour les regressions introduites par cette MAJ : répondre sur ce fil de discussion.

---

Pour chaque anomalie penser à mettre les informations nécessaires pour nous permettre de la reproduire de notre côté (nom du territoire, page, etc.).
Pour faire un retour sur un graphique, se référer aux noms de cette page : https://mondiagartif.beta.gouv.fr/project/51817/all_charts_for_preview

---

Elements à tester :

- [x] Tous les éléments relatifs à la consommation d'espace s'affichent correctement avec le nouveau millésime
- [x] Tous les éléments relatifs à la consommation d'espace présentent des données cohérentes
- [x] L'export complet fonctionne avec le nouveau millésime
- [x] L'export du rapport local fonctionne avec le nouveau millésime

**Comment by inesdartiguenave** [comment]
### Recette commune d'Auch + EPCI Grand Auch (Inès)

**Onglet trajectoires**
Nouveau millésime 2023 absent pour la conso réelle : dans intro en rose; graphique; tableau de données; légende.  
<img width="1200" alt="Capture d’écran 2024-05-08 à 14 00 42" src="https://github.com/MTES-MCT/sparte/assets/84960477/81089169-8f50-4ee2-b7ad-0360effe6d40">
<img width="1258" alt="Capture d’écran 2024-05-08 à 14 02 08" src="https://github.com/MTES-MCT/sparte/assets/84960477/f4ee6a2e-8e8a-471c-ab82-690128c99316">

**Onglet rapport local**
Indicateur 1 pleinement disponible avec le millésime 2023. Retirer la mention "d'ici juin 2024" 
<img width="1250" alt="Capture d’écran 2024-05-08 à 13 54 32" src="https://github.com/MTES-MCT/sparte/assets/84960477/9ecd1377-7d3f-47f4-97c5-1cfe4dbda729">

3ème paragraphe en partant du bas de la page : "pour le premier rapport triennal à produire d'ici août 2024 il **est** possible d'utiliser les fichiers fonciers au 1er janvier 2023. 
<img width="1238" alt="Capture d’écran 2024-05-08 à 13 55 48" src="https://github.com/MTES-MCT/sparte/assets/84960477/98d941b9-93d0-4075-8c5b-0495abc33afa">

**Reply by alexisig** [reply]
Bien vu ! C'est corrigé 👍

**Reply by inesdartiguenave** [reply]
Top avec l'année 2022 ! Dernier détail dans données > source  : millésime max : 2023 (au lieu de 2022) 

<img width="1224" alt="Capture d’écran 2024-05-08 à 18 23 44" src="https://github.com/MTES-MCT/sparte/assets/84960477/78a3d1e7-bd6e-47a6-8b1f-62f2a3b1d08d">

**Comment by inesdartiguenave** [comment]
### Recette SCoT Coeur de Gascogne 

**Onglet consommation**

Graphique "AnnualTotalConsoChart" ne s'affiche pas sur Firefox; ni sur Chrome (mais tableau de données ok) 
<img width="837" alt="Capture d’écran 2024-05-08 à 14 04 23" src="https://github.com/MTES-MCT/sparte/assets/84960477/b5792b5f-87ab-4aa6-8a3f-7c55ff2ce9f4">

**Onglet trajectoires**
Comme pour les mailles communes; EPCI : Nouveau millésime 2023 absent (intro en rose; graphique; tableau de données; légende)

=> Note : j'ai testé le graphique absent de conso sur d'autres territoires; la présence varie : 
- SCoT du Pays d'Issoire Val d'Allier Sud : absent 
- SCoT de la Boucle du Rhône en Dauphiné : présent

**Reply by alexisig** [reply]
Le graphique ne supportait plus la quantité de donnée avec le nouveau millésime ! Je l'ai remplacé par celui qu'on a mis dans l'export du rapport local et ça a l'air de fonctionner à nouveau :) #333

**Reply by inesdartiguenave** [reply]
Oui c'est parfait !

**Comment by inesdartiguenave** [comment]
### Recette exports 

**Rapport local + export complets aux mailles commune + EPCI **
Manque millésime 2023 pour la trajectoire : graphique + tableau de données
Sinon all good : graphiques + tableaux de données + titres ✅

**Mail de téléchargement du rapport local**
Si possible retirer la phrase "Nous nous tenons à votre disposition pour utiliser ce site web financé par le Ministère de la Transition Écologique". Nous trouverons une meilleure formulation. Jérémie étant en congés si tu peux le changer top. Sinon on ferra mercredi.
<img width="678" alt="Capture d’écran 2024-05-08 à 14 36 32" src="https://github.com/MTES-MCT/sparte/assets/84960477/34dd571b-3ca3-4c1c-a571-2b5b42938806">

**Exports complets à l'échelle département + SCoT (word) **

Same pour la trajectoire : millésime 2023 absent du graphique + tableau de données

Débordement pour les tableaux de données associés au graphiques suivants : 
- AnnualConsoComparisonChartExport 
- AnnualConsoByHouseholdChart 
- AnnualPopChartExport
- AnnualHouseholdChartExport

<img width="1043" alt="Capture d’écran 2024-05-08 à 14 18 53" src="https://github.com/MTES-MCT/sparte/assets/84960477/3beed9bb-b333-4be1-94f1-7103d331e082">

<img width="1025" alt="Capture d’écran 2024-05-08 à 14 19 49" src="https://github.com/MTES-MCT/sparte/assets/84960477/b2ef5122-baf8-4eb1-8349-825dc17c0068">

=> Les exports privilégiés étant aux mailles de la commune + EPCI ces débordements de tableaux ne me semblent pas bloquants pour la MEP si @smdsgn peut les résoudre l'amélioration des exports.

**Reply by alexisig** [reply]
Le millésime 2023 concerne les données 2022, du coup je les vois dans les tableaux que tu as joint. Est-ce que c'est bien de ces données la dont tu parles ?

**Reply by alexisig** [reply]
C'est modifié pour le mail (celui-ci du rapport local et celui de l'export complet)

**Reply by inesdartiguenave** [reply]
Le millésime 2023 est bien présent. Ma remarque portaient sur les tableaux qui débordent du cadre / de la page et empêchent parfois de lire le total. C'est plus du design que de la data; et vu uniquement sur les exports de scot + département. ça ne me semble pas bloquant pour la MEP.
